tests/socket-setoption.phpt

```diff
@@ -0,0 +1,20 @@
+--TEST--
+Test of Socket::setOption() with and without parameters
+--FILE--
+<?php
+    $socket = new Socket(\Socket::AF_INET, \Socket::SOCK_STREAM, \Socket::SOL_TCP);
+    $socket->setOption();
+
+    try {
+        $socket->setOption("hello", "world", new stdClass());
+    } catch(Throwable $throwable) {
+        var_dump($throwable->getMessage());
+    }
+    var_dump($socket->setOption(SOL_SOCKET, SO_REUSEADDR, 1));
+    var_dump($socket->getOption(SOL_SOCKET, SO_REUSEADDR));
```

Comment:
same again here - use of global ext/sockets constants causes this to fail without ext/sockets